src/dashboard/src/components/archival_storage/views.py

```diff
@@ -566,6 +568,38 @@ def es_pager(page, page_size):
                   )
 
 
+def _add_encrypted(aips):
+    """Return the ``aips`` list of dicts, adding an 'encrypted' boolean key to
+    each AIP dict indicating whether the AIP is stored encrypted. This is
+    accomplished by making two new requests to the storage service to get the
+    list of locations and spaces and using that information to determine the
+    space of each AIP and whether the AIP is encrypted.
+    """
+    aip_storage_locs = storage_service.get_location(purpose="AS")
+    spaces = storage_service.get_space()
+    for aip in aips:
+        aip['encrypted'] = False
+        location = None
+        space = None
+        file_path = aip['filePath']
+        for loc in aip_storage_locs:
+            loc_path = loc['path']
+            if file_path.startswith(loc_path) and file_path.replace(
+                    loc_path, '', 1).startswith('/'):
+                location = loc
+        if not location:
+            continue
+        for spc in spaces:
+            if spc['uuid'] in location['space']:
+                space = spc
+                break
+        if not space:
+            continue
+        if space['access_protocol'] == 'GPG':
+            aip['encrypted'] = True
```

Comment:
Could we return this property directly from the SS?

Reply:
Yes, and this is essentially what sevein advised too. I'll modify this PR to indicate AIP encryption status in the Elasticsearch index. This way it is computed once and will require no extra requests.